### PR TITLE
Update SA1000 to handle checked operator declarations correctly

### LINKED ISSUE #3478: SA1000 doesn't handle checked operator definitions
C# 11 is introducing user-defined checked operators (`public static T operator checked +(T value);`) and currently triggers `SA1000` saying no space is expected after `checked`.

The analyzer should be updated to allow a space when checked is used in this context, seeing as it differs from the normal `checked(expr)` syntax.

### PR DESCRIPTION
**Pull request by bjornhellander**
Fixes #3478

### CODE REVIEW

**Comment by bjornhellander** [comment]
Any comment on this, @sharwell?

File: StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1000KeywordsMustBeSpacedCorrectly.cs [hunk 1]
```diff
@@ -115,13 +115,22 @@ private static void HandleSyntaxTree(SyntaxTreeAnalysisContext context)
 
                 case SyntaxKind.CheckedKeyword:
                 case SyntaxKind.UncheckedKeyword:
-                    if (token.GetNextToken().IsKind(SyntaxKind.OpenBraceToken))
+                    switch (token.Parent.Kind())
                     {
+                    case SyntaxKind.CheckedStatement:
+                    case SyntaxKind.UncheckedStatement:
+                    case SyntaxKind.OperatorDeclaration:
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1000KeywordsMustBeSpacedCorrectly.cs:122]: ❔ Does this need to support conversion operators as well?

Comment (bjornhellander) [reply StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1000KeywordsMustBeSpacedCorrectly.cs:122]: It does. Thank you! I didn't realize it was a different syntax kind. On its way.

Comment (bjornhellander) [reply StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1000KeywordsMustBeSpacedCorrectly.cs:122]: Updated

**Comment by bjornhellander** [comment]
Any other comments on this, @sharwell? The Roslyn version needed for the tests in this change does not have the formatting problem that 4.4.0 has. Can this get merged before upgrading to a Roslyn with official c# 11 support?

**Comment by bjornhellander** [comment]
@sharwell?

**Review by sharwell — APPROVED** [review]

**Comment by sharwell** [comment]
@bjornhellander this looks good but needs to be rebased to fix a conflict

**Review by sharwell — APPROVED** [review]

**Pull request merged**